homeassistant/components/bkk/sensor.py

```diff
@@ -0,0 +1,314 @@
+        """Initialize the sensor."""
+        self._name = "BKK Data Updater"
+        self.api = futar.BkkFutar()
+        self.routes = None
+        self.stop_id = (
+            f"BKK_{config[CONF_STOP][1:]}"
+            if config[CONF_STOP][0] == "#"
+            else f"BKK_{config[CONF_STOP]}"
+        )
+        self._data = None
+        self._state = None
+        self._icon = ICONS["API"]
+        self.is_grouped = config[CONF_GROUPED]
+        self._only_departures = config[CONF_ONLY_DEPARTURES]
+        self._minutes_after = config[CONF_MINUTES_AFTER]
+
+    def get_data(self):
+        """Fetch data from BKK api."""
+        try:
+            response = self.api.arrivals_and_departures_for_stop(
+                stopid=self.stop_id,
+                only_departures=self._only_departures,
+                minutes_after=self._minutes_after,
+            )
+        except ConnectionError:
+            _LOGGER.error("Unable to fetch data from api.")
+        else:
+            status = response["status"]
+            if status == "OK":
+                self._data = response["data"]
+            self._state = status
+            return self._data
+
+    async def async_added_to_hass(self) -> None:
+        """Create group after entity added."""
+        entity_ids = [ent.entity_id for ent in self.routes]
+        stop_name = self.data["references"]["stops"][self.stop_id]["name"]
+        if self.is_grouped:
+            await async_create_group(self.hass, stop_name, entity_ids)
+
+    @property
+    def state(self):
+        """Get BKKDataUpdater state."""
+        return self._state
+
+    @property
+    def hidden(self) -> bool:
+        """Set BKKDataUpdater hidden."""
+        return True
+
+    @property
+    def name(self):
+        """Get BKKDataUpdater name."""
+        return self._name
+
+    @property
+    def icon(self):
+        """Return BBKKDataUpdater icon."""
+        return self._icon
+
+    def update(self):
+        """Fetch data and update BKKRoute entities."""
+        try:
+            self.get_data()
+        except ConnectionError:
+            _LOGGER.error("Unable to fetch data from BKK api!")
+        else:
+            stop_times = self._data["entry"]["stopTimes"]
+            for i, route in enumerate(self.routes):
+                try:
+                    stop_time = stop_times[i]
+                    route.update_fields(
+                        stop_time,
+                        self._data["references"]["routes"],
+                        self._data["references"]["trips"],
+                    )
+                except IndexError:
+                    route.clear_fields()
+
+    @property
+    def data(self):
+        """Return last fetched data."""
+        return self._data
+
+
+class BkkRoute(Entity):
+    """Class representing a route in a stop/station."""
+
+    def __init__(self, stop_id, idx, vehicle_type):
+        """Initialize the sensor."""
+        self._id = self._line = ""
```

Comment:
Please use None instead of empty strings as default or empty values.

Reply:
fixed